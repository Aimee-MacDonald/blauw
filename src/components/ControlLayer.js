import React from 'react'
import styled from 'styled-components'

import Navigation from './controls/Navigation'
import Toolbox from './controls/Toolbox'

export default class ControlLayer extends React.Component{
  constructor(props){
    super(props)
  }

  render(){
    return(
      <StyledControlLayer>
        <Navigation/>
        <Toolbox/>
      </StyledControlLayer>
    )
  }
}

const StyledControlLayer = styled.div`
  position: fixed;
  z-index: 1;
  width: 100%;
  height: 0;
  overflow: visible;
  display: flex;
  flex-flow: nowrap row;
  justify-content: space-between;
`