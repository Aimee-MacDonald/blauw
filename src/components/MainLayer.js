import React, {useContext, useEffect} from 'react'
import styled from 'styled-components'
import {connect} from 'react-redux'

import {ServerConnectionContext} from '../util/ServerConnection'

import {getMainsFlags} from '../state_management/selectors/navigation'

import BookingSheet from './mains/booking_sheet/BookingSheet'
import Checkout from './mains/Checkout'
import Stock from './mains/Stock'
import Notes from './mains/Notes'
import EditRoomsList from './mains/EditRoomsList'

export const MainLayer = props => {
  const connection = useContext(ServerConnectionContext)

  useEffect(() => {
    connection.dispatch({reducer: 'bookings', action: {type: 'REFRESH_BOOKINGS'}})
  }, [])

  return(
    <StyledMainLayer>
      {props.bookingSheet && <BookingSheet/>}
      {props.checkout && <Checkout/>}
      {props.stock && <Stock/>}
      {props.notes && <Notes/>}
      {props.editRoomsList && <EditRoomsList/>}
    </StyledMainLayer>
  )
}

const StyledMainLayer = styled.div`
  background-color: purple;
  position: fixed;
  z-index: 10;
  width: 100%;
  height: 100%;
`

const mapStateToProps = ({navigation}) => (getMainsFlags(navigation))

export default connect(mapStateToProps)(MainLayer)