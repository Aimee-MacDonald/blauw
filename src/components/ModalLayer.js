import React from 'react'
import styled from 'styled-components'
import {connect} from 'react-redux'

import TempModal from './modals/TempModal'
import CreateBooking from './modals/CreateBooking'

const ModalLayer = props => (
  <StyledModalLayer {...props}>
    <CreateBooking />
  </StyledModalLayer>
)

const StyledModalLayer = styled.div`
  background-color: ${props => props.modalActive ? 'hotpink' : 'transparent'};
  position: fixed;
  z-index: 2;
  width: 100%;
  height: ${props => props.modalActive ? '100%' : '0'};
  overflow: hidden;
  transition: 0.4s;
  display: flex;
  justify-content: center;
  align-items: center;
`

const mapStateToProps = ({modal}) => ({
  modalActive: !!modal.activeModal
})

export default connect(mapStateToProps)(ModalLayer)