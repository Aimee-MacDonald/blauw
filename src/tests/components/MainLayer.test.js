import React from 'react'
import {shallow} from 'enzyme'

import {MainLayer} from '../../components/MainLayer'

test('Render Booking Sheet', () => {
  const wrapper = shallow(<MainLayer
    bookingSheet={true}
    checkout={false}
    stock={false}
    notes={false}
    addRoom={false}
  />)
  
  expect(wrapper.getElement()).toMatchSnapshot()
})

test('Render Checkout', () => {
  const wrapper = shallow(<MainLayer
    bookingSheet={false}
    checkout={true}
    stock={false}
    notes={false}
    addRoom={false}
  />)
  
  expect(wrapper.getElement()).toMatchSnapshot()
})

test('Render Stock', () => {
  const wrapper = shallow(<MainLayer
    bookingSheet={false}
    checkout={false}
    stock={true}
    notes={false}
    addRoom={false}
  />)
  
  expect(wrapper.getElement()).toMatchSnapshot()
})

test('Render Notes', () => {
  const wrapper = shallow(<MainLayer
    bookingSheet={false}
    checkout={false}
    stock={false}
    notes={true}
    addRoom={false}
  />)
  
  expect(wrapper.getElement()).toMatchSnapshot()
})

test('Render Add Room', () => {
  const wrapper = shallow(<MainLayer
    bookingSheet={false}
    checkout={false}
    stock={false}
    notes={false}
    addRoom={true}
  />)
  
  expect(wrapper.getElement()).toMatchSnapshot()
})