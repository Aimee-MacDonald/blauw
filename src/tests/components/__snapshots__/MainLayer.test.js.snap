// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Add Room 1`] = `<styled.div />`;

exports[`Render Booking Sheet 1`] = `
<styled.div>
  <Memo(Connect(BookingSheet)) />
</styled.div>
`;

exports[`Render Checkout 1`] = `
<styled.div>
  <Checkout />
</styled.div>
`;

exports[`Render Notes 1`] = `
<styled.div>
  <Notes />
</styled.div>
`;

exports[`Render Stock 1`] = `
<styled.div>
  <Stock />
</styled.div>
`;
